src/domain/individualHarbor/pierSelectHeader/PierSelectHeader.tsx

```diff
@@ -11,40 +11,52 @@ interface PierSelectHeaderProps {
   readonly piers: Pier[];
   readonly selectedPier?: Pier | null;
   onPierSelect(pier: Pier | null): void;
+  onEdit?(pier: Pier): void;
 }
 
 const PierSelectHeader: React.FC<PierSelectHeaderProps> = ({
   className,
   piers,
   selectedPier,
   onPierSelect,
+  onEdit,
 }) => {
   const { t } = useTranslation();
   return (
     <div className={classNames(styles.container, className)}>
       <div className={styles.selectContainer}>
-        <button
-          onClick={() => onPierSelect(null)}
-          className={classNames(styles.pierButton, {
-            [styles.pierButtonSelected]: !selectedPier,
-          })}
-        >
-          {t('individualHarbor.tableHeaders.all')}
-        </button>
-        {piers.map((pier, idx) => (
+        <div>
           <button
-            key={idx}
-            onClick={() => onPierSelect(pier)}
+            onClick={() => onPierSelect(null)}
             className={classNames(styles.pierButton, {
-              [styles.pierButtonSelected]:
-                selectedPier && selectedPier.identifier === pier.identifier,
+              [styles.pierButtonSelected]: !selectedPier,
             })}
           >
-            {`${t('individualHarbor.tableHeaders.identifier')} ${
-              pier.identifier
-            }`}
+            {t('individualHarbor.tableHeaders.all')}
           </button>
-        ))}
+          {piers.map((pier, idx) => (
```

Comment:
If the pier has an `Id`, it's better to use it as a `key` value instead of the index.

Reply:
Done.